cosmos-sdk-proto/Cargo.toml

```diff
@@ -16,18 +16,22 @@ edition = "2021"
 rust-version = "1.72"
 
 [dependencies]
+ics23 = "0.11.1"
 prost = "0.12"
-prost-types = "0.12"
+prost-wkt-types = "0.5"
```

Comment:
It seems like this crate requires `protoc`?

```
error: failed to run custom build command for `prost-wkt-types v0.5.0`

Caused by:
  process didn't exit successfully: `/home/runner/work/cosmos-rust/cosmos-rust/target/release/build/prost-wkt-types-d2782fb67e78a862/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at 'Could not find `protoc` installation and this build crate cannot proceed without
      this knowledge. If `protoc` is installed and this crate had trouble finding
      it, you can set the `PROTOC` environment variable with the specific path to your
      installed `protoc` binary.If you're on debian, try `apt-get install protobuf-compiler` or download it from https://github.com/protocolbuffers/protobuf/releases
```

As things currently stand, both `tendermint-proto` and `cosmos-sdk-proto` pregenerate all the protos so downstream users don't need `protoc` installed in order to be able to use the crates.

So changing this would make `protoc` a hard dependency where it wasn't before.

Reply:
Ah sorry, didn't realize that as I have protoc installed locally. Let's then stick with `prost-types`.

Reply:
Ah my bad, I should have realized that and asked if it was okay. I actually don't see a good reason for moving to `prost-wkt-types` right now so I will revert that change so that we can keep pre-generating all the protos.

Reply:
Oh now I remember: Because some Protobuf messages contain `Any`, `Duration` or `Timestamp`, we need those so-called "well-known" types to derive `Serialize` in order to derive `Serialize` for the enclosing message via `pbjson`. Unfortunately, `prost-types` does not provide such an impl.

Potential solutions I see:
a) Use `prost-wkt-types` instead of `prost-types` and incur the cost of downstream users having to install `protoc`
b) Use `pbjson-types` instead of `prost-types`
c) Switch between `prost-types` and `pbjson-types` at compile time based on whether the `serde` feature is enabled or not

Would one of those be acceptable or do you see another solution?

Reply:
(d) Instruct `prost-build` to generate Rust definitions for the `google.protobuf` messages (ie. those "well-known types") instead of forwarding them to `prost-types` or another crate and let `pbjson-build` derive the `serde` impls for those.

Reply:
Do any of those have an integration with the `prost::Name` trait like `prost_types::Any::{from_msg, to_msg}` do? We use that a lot in `cosmrs`: https://docs.rs/prost-types/latest/prost_types/struct.Any.html#impl-Any

Reply:
I suppose if we generated our own Rust definitions for the `google.protobuf` messages, we could add `from_msg`/`to_msg` just like `prost-types` does.

Reply:
Perhaps to break this PR up into more manageable chunks, you could PR just this change? (and potentially other changes to `proto-build`)

Reply:
@tony-iqlusion Do we need any well-know types other than ~Duration`, `Timestamp` and `Any` for now?

Reply:
`Duration`, `Timestamp`, and `Any` sound fine to me

Reply:
Okay, we're now using `tendermint_proto::google::protobuf` types which support both `prost::Name` integrations and `informalsystems/pbjson`